pkg/virt-handler/non-root.go

```diff
@@ -106,29 +107,30 @@ func (d *VirtualMachineController) prepareStorage(vmi *v1.VirtualMachineInstance
 	return changeOwnershipOfHostDisks(vmi, res)
 }
 
-func getTapDevices(vmi *v1.VirtualMachineInstance) (map[string]string, error) {
-	macvtap := map[string]struct{}{}
+func getTapDevices(vmi *v1.VirtualMachineInstance, networkBindings map[string]v1.InterfaceBindingPlugin) (map[string]string, error) {
+	tapNetworks := map[string]struct{}{}
+	domainAttachmentByInterfaceName := domainspec.DomainAttachmentByInterfaceName(vmi.Spec.Domain.Devices.Interfaces, networkBindings)
 	for _, inf := range vmi.Spec.Domain.Devices.Interfaces {
-		if inf.Macvtap != nil {
-			macvtap[inf.Name] = struct{}{}
+		if domainAttachmentByInterfaceName[inf.Name] == string(v1.Tap) && inf.Masquerade == nil && inf.Bridge == nil {
+			tapNetworks[inf.Name] = struct{}{}
 		}
 	}
 
 	networkNameScheme := namescheme.CreateHashedNetworkNameScheme(vmi.Spec.Networks)
 	tapDevices := map[string]string{}
 	for _, net := range vmi.Spec.Networks {
-		_, isMacvtapNetwork := macvtap[net.Name]
-		if podInterfaceName, exists := networkNameScheme[net.Name]; isMacvtapNetwork && exists {
+		_, isTapNetwork := tapNetworks[net.Name]
+		if podInterfaceName, exists := networkNameScheme[net.Name]; isTapNetwork && exists {
 			tapDevices[net.Name] = podInterfaceName
```

Comment:
This is odd.

There is a hidden assumption that if there is a device behind a tap domainAttachement, it is always with the pod interface name.
Why is that correct? Maybe it is a device used with the tap* naming.

Well, you could always define this as a "rule", but it feels ugly.
Can this be solved in the future or will we be stuck with this forever? (I mean, assuming we let this slide in, can this be improved without breaking things).

Reply:
yes, this is the rule. I should use the pod interface naming.
If we will improve it in the future we will have to somehow pass the "name" of the tap device.
I don't think it will break anything, we can say that the default is the pod interface name.